queries/c_sharp/highlights.scm

```diff
@@ -9,28 +9,31 @@
 (interpolation) @none
 
 (invocation_expression
-     (member_access_expression
-            name: (identifier) @method))
+  (member_access_expression
+    name: (identifier) @method))
+
+(invocation_expression
+  function: (conditional_access_expression
+    (member_binding_expression
+      name: (identifier) @method)))
 
 (namespace_declaration
   name: [(qualified_name) (identifier)] @namespace)
 
-((member_access_expression
-  (identifier) @type
-  (_) .)
-  (#match? @type "^[A-Z].*[a-z]"))
-
 (qualified_name
   (identifier) @type)
 
 (invocation_expression
       (identifier) @method)
 
-((identifier) @field
- (#match? @field "^_"))
```

Comment:
@svermeulen I generally like the C++ member conventions. Not sure how popular they are in C#

Reply:
All fields that are used within a method without `this.*` can be found in the same file. So we could remove the check since we will be able to find all those with an efficient `is?` in future

Reply:
> All fields that are used within a method without `this.*` can be found in the same file.

Unfortunately, this is not the case. They could be found in another file as part of a `partial class`, `abstract class` or even an "extended" class (overridden might be a better word?).

I guess I could revert this change since an identifier that starts with a `_` should always be a field. It is a convention, but is not enforced at all. I personally encounter both daily.

Reply:
In C++ I made it fused it to a single regex. Decide yourself whether you think this rule should be kept or not.

```
((identifier) @field
 (#match? @field "(^_|^m_|_$)"))
```